disco_aws_automation/disco_vpc_peerings.py

```diff
@@ -224,143 +231,168 @@ def _get_peering_lines():
+
+        if '*' in (unresolved_peering.source_endpoint.type, unresolved_peering.target_endpoint.type):
+            raise VPCConfigError('Wildcards are not allowed for VPC type in "%s". '
+                                 'Please specify a VPC type when using a wild card for the VPC name' % line)
+
+        def resolve_endpoint(endpoint):
+            """
+            Convert a PeeringEndpoint that may contain wildcards into a list of PeeringEndpoints
+            with wildcards resolved
+            """
+            endpoints = []
+            for vpc in existing_vpcs:
+                tags = tag2dict(vpc['Tags'])
+                if endpoint.name in ('*', tags['Name']) and endpoint.type == tags['type']:
+                    endpoints.append(PeeringEndpoint(tags['Name'], tags['type'], endpoint.metanetwork, vpc))
+            return endpoints
+
+        # find the VPCs that match the peering config. Replace wildcards with real VPC names
+        source_endpoints = resolve_endpoint(unresolved_peering.source_endpoint)
+        target_endpoints = resolve_endpoint(unresolved_peering.target_endpoint)
+
+        # generate new connection lines by peering the cross product of every source and target endpoint
+        return {PeeringConnection(peering[0], peering[1])
+                for peering in product(source_endpoints, target_endpoints)
+                # Don't peer a VPC with itself
+                if not peering[0] == peering[1]}
+
+
+class PeeringEndpoint(object):
+    """
+    Represents one side of a PeeringConnection
+    """
+    def __init__(self, env_name, env_type, metanetwork, vpc=None):
+        self.name = env_name
+        self.type = env_type
+        self.metanetwork = metanetwork
+        self.vpc = vpc
+
+    def __eq__(self, other):
+        return hash(self) == hash(other)
+
+    def __hash__(self):
+        return hash(str(self))
+
+    def __str__(self):
+        return '%s:%s/%s' % (
+            self.name,
+            self.type,
+            self.metanetwork
+        )
+
+
+class PeeringConnection(object):
+    """
+    Represents a connection between two different VPCs
+    """
+    def __init__(self, source_endpoint, target_endpoint):
+        """
+        Args:
+            source_endpoint (PeeringEndpoint): source side of connection
+            target_endpoint (PeeringEndpoint): target side of connection
+        """
+        self.source_endpoint = source_endpoint
+        self.target_endpoint = target_endpoint
+
+    def contains_vpc_name(self, vpc_name):
+        """ Return true if the given vpc_name is the name of a VPC on one of the sides of the connection"""
+        return self.source_endpoint.name == vpc_name or self.target_endpoint.name == vpc_name
+
+    def contains_vpc_id(self, vpc_id):
+        """ Return true if the given vpc_id is the id of a VPC on one of the sides of the connection"""
+        return self.source_endpoint.vpc['VpcId'] == vpc_id or self.target_endpoint.vpc == vpc_id
```

Comment:
Shouldn't it be `self.target_endpoint.vpc['VpcId'] == vpc_id` after the `or` operator?